engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
getNodeId() returning null should be handled as a separate JIRA and a separate PR. For this PR, I'd suggest focusing on a known issue (NPE in StreamingAppMasterService.java line 1014).